### ISSUE TITLE
More 500 Server Errors on Install

### ISSUE THREAD

**Issue by marcodeluca** [JC5/firefly-iii#296]
Hey there, excited to try out firefly-iii but having issues with the install. Followed your instructions exactly on a clean VM, I gone through other similar issues (like https://github.com/JC5/firefly-iii/issues/277) but nothing resolved my issue. It looks like I've got a problem specifically related to MailError.php.

Here is the apache2 error log. Maybe something with the mail config is throwing it off. i used the standard GMAIL server settings in the config setup. Let me know if you have any ideas.

[Thu Aug 25 20:33:50.745527 2016] [:error] [pid 23815] [client 174.118.90.133:47910] PHP Fatal error:  Uncaught exception 'ErrorException' with message 'Argument 2 passed to FireflyIII\\Jobs\\MailError::__construct() must be an instance of FireflyIII\\Jobs\\string, string given, called in /var/www/html/firefly-iii/app/Exceptions/Handler.php on line 95 and defined' in /var/www/html/firefly-iii/app/Jobs/MailError.php:50\nStack trace:\n#0 /var/www/html/firefly-iii/app/Jobs/MailError.php(50): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleError(4096, 'Argument 2 pass...', '/var/www/html/f...', 50, Array)\n#1 /var/www/html/firefly-iii/app/Exceptions/Handler.php(95): FireflyIII\\Jobs\\MailError->__construct(Array, 'marco@fractionl...', '174.118.90.133', Array)\n#2 /var/www/html/firefly-iii/bootstrap/cache/compiled.php(2163): FireflyIII\\Exceptions\\Handler->report(Object(Symfony\\Component\\Debug\\Exception\\FatalErrorException))\n#3 /var/www/html/firefly-iii/bootstrap/cache/compiled.php(2181): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleException(Object(Symfony\\Component\\Debug\\Exception\\FatalErrorException))\n#4 in /var/www/html/firefly-iii/app/Jobs/MailError.php on line 50

[Thu Aug 25 20:33:50.744347 2016] [:error] [pid 23815] [client 174.118.90.133:47910] PHP Warning:  Unsupported declare 'strict_types' in /var/www/html/firefly-iii/app/Models/Account.php on line 10

[Thu Aug 25 20:33:50.744591 2016] [:error] [pid 23815] [client 174.118.90.133:47910] PHP Parse error:  syntax error, unexpected ':', expecting ';' or '{' in /var/www/html/firefly-iii/app/Models/Account.php on line 167

**Comment by JC5** [comment]
Firefly III requires PHP 7.0 or higher, and your errors indicate you run something else. Please check the output of the ``phpinfo() ;`` command or run ``php -v`` to verify you are running the correct version.

**Comment by marcodeluca** [comment]
$ php -v
PHP 7.0.10-1+deb.sury.org~trusty+1 (cli) ( NTS )
Copyright (c) 1997-2016 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2016 Zend Technologies
    with Zend OPcache v7.0.10-1+deb.sury.org~trusty+1, Copyright (c) 1999-2016, by Zend Technologies

**Comment by JC5** [comment]
That does not make sense at all.

- The string error is a php5 error because php5 cannot handle string as a argument type.
- The strict_types error is a php5 error because php5 does not support the declare thing.
- The  Account.php error is a php5 error because php5 does not support return types.

I am not sure how I can help you.

**Comment by JC5** [comment]
Are you sure that your webserver and command line php version are the same? It could be different, especially when you install php7 using external repositories.

**Comment by xzaz** [comment]
Create a php info page check the php version there.

**Comment by marcodeluca** [comment]
the server WAS running php5, was due to some php5 modules being enabled that had apache running with php5 even though I had installed php7 as per your instructions. In case anyone else has this issue with updating php from 5 to 7, don't use php -v put:

    <?php
      phpinfo();
    ?>

in your root folder, see what it says, and if php5 is still running, a quick google search will give you instructions on how to make sure it's disabled.

@JC5  and @xzaz that's for the input!
